state_synchronizer/src/peer_manager.rs

```diff
@@ -135,11 +149,33 @@ impl PeerManager {
         self.peers
             .iter()
             .filter(|&(_, peer_info)| peer_info.is_alive && peer_info.is_upstream)
-            .map(|(peer_id, peer_info)| (peer_id, peer_info))
             .collect()
     }
 
     pub fn get_network_sender(&self, peer_id: &PeerId) -> Option<StateSynchronizerSender> {
         self.network_senders.get(peer_id).cloned()
     }
+
+    pub fn update_timeout_request(&mut self, version: u64, peer_id: Option<PeerId>) {
```

Comment:
I'd prefer to have 2 separate methods. `process_request` and `process_response`. Although they migh have similar functionality now, it'd be handy to eventually differentiate them